azuread/configuration.go

```diff
@@ -164,16 +175,22 @@ func (p *azureFedAuthConfig) provideActiveDirectoryToken(ctx context.Context, se
 	case ActiveDirectoryServicePrincipal, ActiveDirectoryApplication:
 		switch {
 		case p.certificatePath != "":
-			cred, err = azidentity.NewClientCertificateCredential(tenant, p.clientID, p.certificatePath, &azidentity.ClientCertificateCredentialOptions{Password: p.clientSecret})
+			certData, err := os.ReadFile(p.certificatePath)

```

Comment:
Is this change required by the azidentity upgrade? 
Did `azidentity.NewClientCertificateCredential` in the prior version validate the path in the constructor? If not this change introduces a subtle shift in where the error gets raised if the path isn't valid. 
It'd be good to have a test showing what errors are expected for an invalid cert path or invalid client secret

Reply:
yes, this is required for any azidentity version above 0.11 -- from their changelog

```
NewClientCertificateCredential requires a []*x509.Certificate and crypto.PrivateKey instead of a path to a certificate file. Added ParseCertificates to simplify getting these in common cases:
// before
cred, err := NewClientCertificateCredential("tenant", "client-id", "/cert.pem", nil)

// after
certData, err := os.ReadFile("/cert.pem")
certs, key, err := ParseCertificates(certData, password)
cred, err := NewClientCertificateCredential(tenantID, clientID, certs, key, nil)
```